---
title: "DGE_Analysis"
author: "Jonathan Anzules"
output: html_document
date: "2023-06-09"
---

**Introduction**

This file does a couple of things:
 - Prepares all of the relevant metadata data into a table
 - Prepares a transcript to gene mapping of for importing salmon files
 - tximport - imports the salmon quant.sf files into count matrix needed for DESeq
 - deseq - Next we used DESeq2 to do the differential gene expression analysis
 - Results from deseq is used for the Gene Set Enrichment Analysis 


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

setwd("/Users/jonan/Documents/Tyseq/Code/") #All file locations will have a relative directionality in the file tree. 
library(tximport) # Importing salmon files and preparing for deseq
library(DESeq2) # DGE Analysis
library(GenomicFeatures) # Generating the tx2gene 
library(ggplot2)
library(tidyr)
library(tidyverse)
library(fgsea) # pathway enrichment analysis
library(biomaRt) # Converting ensembleID to geneID
library(data.table)
library(plotly)
library(dplyr)
```
# Data Prep
```{r tx2gene}
txdb <- makeTxDbFromGFF("../Data/hg19/gencode.v19.chr_patch_hapl_scaff.annotation.gtf")

# keytypes(txdb)
k <- keys(txdb, keytype = "TXNAME")

tx2gene <- select(txdb, k, "GENEID", "TXNAME") # only works om 4.3.0 # Restart and clear output if this doesn't work
tx2gene <- as.data.frame(tx2gene)
tx2gene <- tx2gene[, c("TXNAME", "GENEID")]

```

```{r sample setup}
# Setting up files for upload

# "../Data/SalmonQuant/quantALL/"
control_files = c("../Data/SalmonQuant/quantALL/1_XIN460_NHI_ATTACT_L005_R1_001_quant.sf", 
                  "../Data/SalmonQuant/quantALL/2_XGM061_NHI_TCCGGA_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/3_XJL334_NHI_CGCTCA_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/4_ZCA126_NHI_GAGATT_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/5_XGZ492_NHI_ATTCAG_L005_R1_001_quant.sf")

T1D_files = c("../Data/SalmonQuant/quantALL/6_AAFS251_T2DHI_GAATTC_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/7_AAJ2482_T2DHI_CTGAAG_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/8_AABW178_T2DHI_TAATGC_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/9_XIX456_T2DHI_CGGCTA_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/10_ABDG032_T2DHI_TCCGCG_L005_R1_001_quant.sf")

# Creating a sample table that maps each sample to its corresponding quant.sf file
# _C = control
# _D = Diabetic
sample_table <- data.frame(
  sampleName = c("XIN460_C", "XGM061_C", "XJL334_C", "ZCA126_C", "XGZ492_C",
                 "AAFS251_D", "AAJ2482_D", "AABW178_D", "XIX456_D", "ABDG032_D"),
  fileName = c(control_files, T1D_files),
  diabetes_status = c(rep("No", 5), rep("Yes", 5))
)

```

```{r Loading Count Matrix and Filtering}
# import salmon abundance counts into an r data ckass that can be used by desseq
txi <- tximport(files = sample_table$fileName, type = "salmon", tx2gene = tx2gene, ignoreAfterBar = TRUE) 
counts <- round(txi$counts)

# Calculate the number of samples in which each gene is expressed for each group
expressed_control <- rowSums(counts[, sample_table$diabetes_status == "No"] > 0)
expressed_diabetes <- rowSums(counts[, sample_table$diabetes_status == "Yes"] > 0)

# Filter genes based on the 80% criterion for each group
keep_genes <- (expressed_control >= 0.8 * sum(sample_table$diabetes_status == "No")) | 
              (expressed_diabetes >= 0.8 * sum(sample_table$diabetes_status == "Yes"))

# Apply the filter to the counts matrix
filtered_counts <- counts[keep_genes, ]


```


```{r DESeq differential gene expression analysis}

columnData <- data.frame(condition = sample_table$diabetes_status)

dds <- DESeqDataSetFromMatrix(countData = filtered_counts, colData = columnData, design = ~condition) # ~condition is what tells it that I am interested in seeing the relationships based on condition

#Adding the right column names to the dds object
colnames(dds) <- sample_table$sampleName
# dds$condition - These should match with colnames(dds); they do.

## Normalizing the counts

# Estimated size factors are determined by the library size
dds <- estimateSizeFactors(dds)

# Differential gene expression analysis
dds <- DESeq(dds)

# Getting the results based on the condition of diabetes or not
results <- results(dds, contrast = c("condition", "Yes", "No"))

```

```{r - DESeqDataSet to tabular data}

# Extracting expression values from DESeqDataSet
expression_values <- assay(dds, "counts")

# Creating a dataframe
gene_expression_data <- as.data.frame(expression_values)
gene_expression_data$EnsembleID <- rownames(gene_expression_data)

# Removing the version number from EnsembleID using regex
gene_expression_data$EnsembleID <- gsub("\\.\\d+$", "", gene_expression_data$EnsembleID)

## Obtaining gene ID

# Connect to the Ensembl database using biomaRt
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

# Convert Ensemble IDs to ensembl_gene_id Gene IDs
converted_ids <- getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                       filters = "ensembl_gene_id",
                       values = gene_expression_data$EnsembleID,
                       mart = ensembl)

# Merge the converted IDs with your gene expression data
gene_expression_data <- merge(gene_expression_data, converted_ids, by.x = "EnsembleID", by.y = "ensembl_gene_id")

# Removing empty and NA genesymbol entries
gene_expression_data <- subset(gene_expression_data, !(is.na(external_gene_name) | external_gene_name == ""))

write.csv(gene_expression_data, "../Data/ExpressionData_filtered.csv")


```



```{r Sanity check}
# Checking to see that the right columns are associated to the right patient ID

# Get the column names from DESeq object
col_names = colnames(dds)

# Get the patient IDs from the sample table
patient_ids <- sample_table$sampleName

# Compare the column names with the sample name
# If match 
match_result <- match(col_names, patient_ids)

match_result

# Print the results
for (i in seq_along(col_names)) {
  cat("Column:", col_names[i], " - SampleName ID:", patient_ids[match_result[i]], "\n")
}

```
# Exploring Results
```{r TODO: Add the gene symbols to the results df}

```


```{r top10}
# #Normalizing the counts

top_genes <- results[order(results$padj), ]
top_10_genes <- head(top_genes, n=10)

top_10_genes

# write.csv(top_10_genes, file = "../Data/to_10_genes.csv")


```


```{r top30}

top_genes <- results[order(results$padj), ]
top_30_genes <- head(top_genes, n=30)

top_30_genes

write.csv(top_30_genes, file = "../Data/top_30_genes_2.csv")
```

```{r p-values vs log2foldchange}
# Checking to see the what differences is if I sort by smallest p-values and the
# greatest absolute value in log2foldchange

results

# Sort results by absolute log2FC in descending order
sorted_results <- results[order(abs(results$log2FoldChange), decreasing = TRUE), ]

# Print the sorted results
print(sorted_results)

```

```{r - Identifying Differences}
# I updated salmon to the most up to date version and used the gcbias flag - _2

library(dplyr)
dfA = read.csv("../Data/top_30_genes.csv")
dfB = read.csv("../Data/top_30_genes_2.csv")


# Find values in dfA that are not present in dfB
values_only_in_dfA <- anti_join(dfA, dfB, by = "EnsembleID")

# Find values in dfB that are not present in dfA
values_only_in_dfB <- anti_join(dfB, dfA, by = "EnsembleID")

values_only_in_dfA
values_only_in_dfB

```
#Volcano plot

```{r Volcano plot with selecting labels}

# Sort by adjusted p-values and get top three entries
top_three <- results_df[order(results_df$padj, na.last = NA)[1:3],]

# Create a volcano plot
ggplot(results_df, aes(x = log2FoldChange, y = neg_log10_padj)) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results_df, padj < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  # Adding the text labels for the top three points
  geom_text(data = top_three, aes(label = rownames(top_three)), 
            vjust = -1, hjust = 1) +
  labs(x = "log2 Fold Change", y = "-log10(adjusted p-value)",
       title = "Volcano Plot") +
  theme_bw()


```


```{r -  Plot No label - adjusted p value}
# Results generated in Data Prep
results_df <- as.data.frame(results)

# Calculate the negative log10 of the adjusted p-values
results_df$neg_log10_padj <- -log10(results_df$padj)

# Create a volcano plot
ggplot(results_df, aes(x = log2FoldChange, y = neg_log10_padj)) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results_df, padj < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  labs(x = "log2 Fold Change", y = "-log10(adjusted p-value)",
       title = "Volcano Plot") +
  theme_bw()

top_genes <- results_df[order(results_df$padj), ]
top_10_genes <- head(top_genes, n=10)

write.csv(top_genes, "../Data/topDGEGenes_filtered.csv")

```

```{r -  Plot No label - p value}
# Results generated in Data Prep
results_df <- as.data.frame(results)

# Calculate the negative log10 of the adjusted p-values
results_df$neg_log10_pvalue <- -log10(results_df$pvalue)

# Create a volcano plot
ggplot(results_df, aes(x = log2FoldChange, y = neg_log10_pvalue)) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results_df, pvalue < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  labs(x = "log2 Fold Change", y = "-log10(adjusted p-value)",
       title = "Volcano Plot") +
  theme_bw()
```

```{r - DEG labeled}
library(ggplot2)

# Convert DESeqResults to a data frame
results_df <- as.data.frame(results)

results_df$EnsembleID <- rownames(results_df)
# Create the volcano plot
volcano_plot <- ggplot(results_df, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point() +
  xlim(c(-max(abs(results_df$log2FoldChange)), max(abs(results_df$log2FoldChange)))) +
  ylim(c(0, max(-log10(results_df$padj)))) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  xlab("log2 Fold Change") +
  ylab("-log10(adjusted p-value)") +
  ggtitle("Volcano Plot")

# Add labels to significantly differentially expressed genes
signif_labels <- subset(results_df, padj < 0.05)
volcano_plot_with_labels <- volcano_plot +
  geom_text(data = signif_labels, aes(label = EnsembleID), vjust = 0, hjust = 0)

# Print the volcano plot with labels
print(volcano_plot_with_labels)


```

```{r labeled plot}
library(ggplot2)
library(broom)
library(plotly)

# Convert DESeqResults object to data frame
results <- as.data.frame(results)

# Convert DESeqResults to a data frame using tidy()
# results <- tidy(deseq_results)

# Add EnsembleID as a new column
results$EnsembleID <- rownames(results)

# Create the volcano plot using ggplot2
volcano_plot <- ggplot(results, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point() +
  xlim(c(-max(abs(results$log2FoldChange)), max(abs(results$log2FoldChange)))) +
  ylim(c(0, max(-log10(results$padj)))) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  xlab("log2 Fold Change") +
  ylab("-log10(adjusted p-value)") +
  ggtitle("Volcano Plot")

# Add labels to significantly differentially expressed genes
signif_labels <- subset(results, padj < 0.05)
volcano_plot_with_labels <- volcano_plot +
  geom_text(data = signif_labels, aes(label = EnsembleID), vjust = 0, hjust = 0)

# Convert ggplot to a plotly object
interactive_volcano_plot <- ggplotly(volcano_plot_with_labels)

# Print the interactive volcano plot
print(interactive_volcano_plot)


```

```{r}

# Convert 'padj' column to numeric
results$padj <- as.numeric(results$padj)

# Create the interactive volcano plot with hover text
volcano_plot <- plot_ly(data = results, 
                        x = ~log2FoldChange, 
                        y = -log10(results$padj), # No ~ symbol for log10() function
                        text = ~EnsembleID,
                        hoverinfo = "text",
                        type = "scatter",
                        mode = "markers",
                        marker = list(
                          size = 10,
                          color = ifelse(results_df$padj < 0.05, "red", "blue"),
                          opacity = 0.7
                        )) %>%
  layout(title = "Volcano Plot",
         xaxis = list(title = "log2 Fold Change"),
         yaxis = list(title = "-log10 p-value"))

# Save the interactive plot as an HTML file
htmlwidgets::saveWidget(volcano_plot, file = "../Figures/interactive_volcano_plot.html")


```

```{r html figure}
# Load necessary libraries
library(plotly)
library(dplyr)

# Adjusted p-value cutoff for highlighting most differentially expressed genes
pvalue_cutoff <- 0.05

# Create a new column to indicate significance based on adjusted p-value
results$significant <- ifelse(results$padj <= pvalue_cutoff, "Yes", "No")

# Create the volcano plot with interactive tooltips for Ensemble ID
volcano_plot <- plot_ly(data = results, x = ~log2FoldChange, y = ~-log10(padj), 
                        color = ~significant, colors = list("Yes" = "red", "No" = "black")) %>%
  add_markers(size = 5, alpha = 0.6, hoverinfo = "text",
              text = ~paste("Ensemble ID: ", EnsembleID, "<br>",
                            "log2 Fold Change: ", log2FoldChange, "<br>",
                            "Adjusted p-value: ", padj)) %>%
  layout(xaxis_title = "log2 Fold Change",
         yaxis_title = "-log10(Adjusted p-value)",
         title = "Volcano Plot",
         hovermode = "closest")


plot_ly(data = results, x = ~log2FoldChange, y = ~-log10(padj), 
                        color = ~significant, colors = list("Yes" = "red", "No" = "black")) %>%
  add_markers(size = 5, alpha = 0.6, hoverinfo = "text",
              text = ~paste("Ensemble ID: ", EnsembleID, "<br>",
                            "log2 Fold Change: ", log2FoldChange, "<br>",
                            "Adjusted p-value: ", padj)) %>%
  layout(xaxis_title = "log2 Fold Change",
         yaxis_title = "-log10(Adjusted p-value)",
         title = "Volcano Plot",
         hovermode = "closest")


# Convert plot_ly object to plotly object
plotly_obj <- plotly_build(volcano_plot)

# Convert to JSON format
plotly_json <- plotly:::plotly_json(plotly_obj, FALSE)

# Save the plot as an interactive HTML file
output_file <- "../Figures/interactive_volcano_plot2.html"
htmlwidgets::saveWidget(list(data = plotly_json), file = output_file)


```

```{r}
# Adjusted p-value cutoff for highlighting most differentially expressed genes
pvalue_cutoff <- 0.05

# Create a new column to indicate significance based on adjusted p-value
results$significant <- ifelse(results$padj < pvalue_cutoff, "Yes", "No")

# Create the volcano plot with interactive tooltips for Ensemble ID
# There is something wrong with the color selection
volcano_plot <- plot_ly(data = results, x = ~log2FoldChange, y = ~-log10(padj), 
                        color = ~significant, colors = list("Yes" = "red", "No" = "black")) %>%
  add_markers(size = 5, alpha = 0.6, hoverinfo = "text",
              text = ~paste("Ensemble ID: ", EnsembleID, "<br>",
                            "log2 Fold Change: ", log2FoldChange, "<br>",
                            "Adjusted p-value: ", padj)) %>%
  layout(xaxis_title = "log2 Fold Change",
         yaxis_title = "-log10(Adjusted p-value)",
         title = "Volcano Plot",
         hovermode = "closest")

# Save the plot as an interactive HTML file
output_file <- "../Figures/interactive_volcano_plot2.html"
htmlwidgets::saveWidget(volcano_plot, file = output_file)

```

#Pathway Enrichment analysis (GSEA)

## Cleaning and Preparing data
```{r ranka and dfGSEA generation}
results_df <- as.data.frame(results)

# Removing rows with na values in them
results_clean <- results_df %>% drop_na(log2FoldChange, pvalue)

# Calculating gene rankings
results_clean <- results_clean %>% mutate(rank = log2FoldChange * -log10(pvalue))

# Creating a new dataframe with the necessary columns
dfGSEA <- results_clean %>%
  rownames_to_column(var = "Gene_names") %>%
  select(Gene_names, log2FC = log2FoldChange, pvalue, rank)

# Removing the version number from EnsembleID using regex
dfGSEA$Gene_names <- gsub("\\.\\d+$", "", dfGSEA$Gene_names)

write.csv(dfGSEA, "../Data/DGEdataframe_filtered.csv")
```

## Converting EnsembleID to gene ID

```{r Batch processing of ensembleID}

# Set up the connection to the Ensembl database
ensembl <- useEnsembl(biomart = "ensembl", dataset = "hsapiens_gene_ensembl")

# Batch processing: Retrieve gene symbols for all ensemble IDs in one call
gene_symbols <- getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                      filters = "ensembl_gene_id",
                      values = dfGSEA$Gene_names,
                      mart = ensembl)

# Merge the gene symbols into the original dfGSEA dataframe based on ensemble IDs
dfGSEA <- merge(dfGSEA, gene_symbols, by.x = "Gene_names", by.y = "ensembl_gene_id", all.x = TRUE)

# Rename the merged column to "Gene_symbols"
colnames(dfGSEA)[colnames(dfGSEA) == "external_gene_name"] <- "Gene_symbols"

# Removing empty and NA genesymbol entries
dfGSEA <- subset(dfGSEA, !(is.na(Gene_symbols) | Gene_symbols == ""))

```

```{r Saving DGE Results}
#Saving the results top 50
sigGenes <- subset(dfGSEA, pvalue <= 0.05)
sigGenes <- sigGenes[order(sigGenes$pvalue),]

write.csv(sigGenes, file = "../Data/SigGenes_filtered.csv")


```



## Gene Ontology enrichment analysis
```{r picking top 30}
# Extract gene rank and list in decreasing order for GSEA
ranks <- as.numeric(dfGSEA$rank)
names(ranks) <- dfGSEA$Gene_symbols
ranks <- sort(ranks, decreasing=TRUE)
ranks[1:30]
```

```{r loading gmt file}
# Import .gmt file with pathway information by gene symbol (Gene sets derived from the GO Biological Process ontology)
## Obtained from GSEA 3-6-2023 http://www.gsea-msigdb.org/gsea/msigdb/collections.jsp
go <- gmtPathways("C:/Users/jonan/Documents/Tyseq/Data/c5.go.v2023.1.Hs.symbols.gmt")
```

```{r pathfgsea}
set.seed(06212022)

fgseaRes <- fgsea(pathways = go, 
                  stats = ranks,
                  minSize=6,
                  maxSize=500,
                  nproc=1)
# There are dubplicate gene names
```

# Plotting Pathway Enrichment Analysis

## Picking the top pathways

```{r Saving and Results}
#ssorting by pvalue
fgseaRes <- fgseaRes[order(fgseaRes$pval), ]

# Convert the lists to comma-separated strings
fgseaRes$leadingEdge <- sapply(fgseaRes$leadingEdge, paste, collapse = ",")

# Save the modified dataframe to a CSV file
write.csv(fgseaRes, "C:/Users/jonan/Documents/Tyseq/Data/fgseaResults_with_strings_filtered.csv", row.names = FALSE)

# Read back in the table 
EAResults <- read.csv("C:/Users/jonan/Documents/Tyseq/Data/fgseaResults_with_strings.csv")

```

```{r Picking Top and up/down pathways - pvalue}

# Select top 10 up and down-regulated pathways
topup <- EAResults %>% filter(ES > 0)
topup <- topup[order(topup$pval),]
topup <- topup[1:10,]

topdown <- EAResults %>% filter(ES < 0)
topdown <- topdown[order(topdown$pval),]
topdown <- topdown[1:10,]

top <- rbind(topup, rev(topdown))

# Clean up pathway names
top$pathway <- top$pathway %>% str_replace("GO.+?_", "") %>% str_replace_all("_", " ")

top <- top[order(top$pval),]
top <- top %>% filter(pval<=0.05) %>% filter(!is.na(pathway))


# Write cleaned table
write_csv(top, "C:/Users/jonan/Documents/Tyseq/Data/Top_GSEA_Pathways_GO_filtered.csv")

```

```{r Picking Top and up/down pathways - padj}

# Select top 10 up and down-regulated pathways
topup <- EAResults %>% filter(ES > 0)
topup <- topup[order(topup$padj),]
topup <- topup[1:10,]

topdown <- EAResults %>% filter(ES < 0)
topdown <- topdown[order(topdown$padj),]
topdown <- topdown[1:10,]

top <- rbind(topup, rev(topdown))

# Clean up pathway names
top$pathway <- top$pathway %>% str_replace("GO.+?_", "") %>% str_replace_all("_", " ")

top <- top[order(top$padj),]
top <- top %>% filter(padj<=0.1) %>% filter(!is.na(pathway))


# Write cleaned table
write_csv(top, "C:/Users/jonan/Documents/Tyseq/Data/Padj_Top_GSEA_Pathways_GO.csv")

```

## Ploting Results
```{r sigpathgraphcell, fig.width=11, fig.height=3}
# Summarize pathway information and create negative log p value variable for graphing
pathg <- top %>% filter(pval <= 0.1) 
pathg <- pathg %>% mutate(neglogpvalue = -log10(pval))

if (nrow(pathg) >= 1) {
  # Graph pathways by p value
  pathfig <- ggplot(pathg, aes(x = reorder(pathway, neglogpvalue), y = neglogpvalue)) +
    geom_bar(stat = "identity", fill = "black") +
    coord_flip() +
    scale_x_discrete(name = "Pathways Associated with Diabetes") +
    ylab("-log(p value)") +
    theme(axis.text.x = element_text(face = "bold", size = 10, angle = 0),
          axis.text.y = element_text(face = "bold", size = 10, angle = 0))

  pathfig

  # Save the figure with increased width (e.g., width = 10 inches)
  ggsave("C:/Users/jonan/Documents/Tyseq/Figures/pathsigfig_cts_filtered.png", width = 12)
}

```
```{r Plotting pathways - Red = downregulated}

# Summarize pathway information and create negative log p value variable for graphing
pathg <- top %>% filter(pval <= 0.1) 
pathg <- pathg %>% mutate(neglogpvalue = -log10(pval))

if (nrow(pathg) >= 1) {
  # Graph pathways by p value
  pathfig <- ggplot(pathg, aes(x = reorder(pathway, neglogpvalue), y = neglogpvalue)) +
    # Conditionally set fill color based on the "ES" value
    geom_bar(aes(fill = ifelse(ES < 0, "red", "black")), stat = "identity") +
    coord_flip() +
    scale_x_discrete(name = "Pathways Associated with Diabetes") +
    scale_fill_identity() +  # Ensure the colors are applied as specified
    ylab("-log(p value)") +
    theme(axis.text.x = element_text(face = "bold", size = 10, angle = 0),
          axis.text.y = element_text(face = "bold", size = 10, angle = 0))

  pathfig

  # Save the figure with increased width (e.g., width = 10 inches)
  ggsave("C:/Users/jonan/Documents/Tyseq/Figures/pathsigfig_cts_filtered_redown.png", width = 12)
}


```

  
#### Top 10 pathways
```{r pathgraphcell, fig.width=11, fig.height=6}
# Summarize pathway information and create negative log p value variable for graphing
pathg <- top[1:10,]
pathg <- pathg %>% filter(!is.na(pathway))

pathg <- pathg %>% mutate(neglogpvalue=-log10(pval))

# Graph pathways by p value
pathfig <- ggplot(pathg, aes(x=reorder(pathway, neglogpvalue), y=neglogpvalue)) +
  geom_bar(stat = "identity", fill="black") +
  coord_flip() +
  scale_x_discrete(name="Pathways Associated with Diabetes") +
  ylab("-log(p value)") +
  theme(axis.text.x = element_text(face="bold", size=10, angle=0),
        axis.text.y = element_text(face="bold", size=10, angle=0))

pathfig

ggsave("C:/Users/mecho/Documents/Hertzberg_Lab/T2D/Proteomics_for_Vicki/Analysis_Houser/Pathways/TenPathfig_cts.png")
```

































