---
title: "DGE_Analysis"
output: html_document
date: "2023-06-09"
---

**Introduction**

This file does a couple of things:
 - Prepares all of the relevant metadata data into a table
 - Prepares a transcript to gene mapping of for importing salmon files
 - tximport - imports the salmon quant.sf files into count matrix needed for DESeq
 - deseq - Next we used DESeq2 to do the differential gene expression analysis 

Future Steps
Automate the acquisition of gene descriptions based on the gene ID


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

setwd("/Users/jonan/Documents/Tyseq/Code/") #All file locations will have a relative directionality in the file tree. 
library(tximport) # Importing salmon files and preparing for deseq
library(DESeq2) # DGE Analysis
library(GenomicFeatures) # Generating the tx2gene 

```

```{r tx2gene}

txdb <- makeTxDbFromGFF("../Data/hg19/gencode.v19.chr_patch_hapl_scaff.annotation.gtf")
# keytypes(txdb)
k <- keys(txdb, keytype = "TXNAME")
tx2gene <- select(txdb, k, "GENEID", "TXNAME")
tx2gene <- as.data.frame(tx2gene)
tx2gene <- tx2gene[, c("TXNAME", "GENEID")]

```

```{r sample setup}
# Setting up files for upload
# "../Data/SalmonQuant/quantALL/"
control_files = c("../Data/SalmonQuant/quantALL/1_XIN460_NHI_ATTACT_L005_R1_001_quant.sf", 
                  "../Data/SalmonQuant/quantALL/2_XGM061_NHI_TCCGGA_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/3_XJL334_NHI_CGCTCA_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/4_ZCA126_NHI_GAGATT_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/5_XGZ492_NHI_ATTCAG_L005_R1_001_quant.sf")

T1D_files = c("../Data/SalmonQuant/quantALL/6_AAFS251_T2DHI_GAATTC_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/7_AAJ2482_T2DHI_CTGAAG_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/8_AABW178_T2DHI_TAATGC_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/9_XIX456_T2DHI_CGGCTA_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/10_ABDG032_T2DHI_TCCGCG_L005_R1_001_quant.sf")

# Creating a sample table that maps each sample to its corresponding quant.sf file
# _C = control
# _D = Diabetic
sample_table <- data.frame(
  sampleName = c("XIN460_C", "XGM061_C", "XJL334_C", "ZCA126_C", "XGZ492_C",
                 "AAFS251_D", "AAJ2482_D", "AABW178_D", "XIX456_D", "ABDG032_D"),
  fileName = c(control_files, T1D_files),
  diabetes_status = c(rep("No", 5), rep("Yes", 5))
)

```


```{r Count Matrix}
txi <- tximport(files = sample_table$fileName, type = "salmon", tx2gene = tx2gene, ignoreAfterBar = TRUE) 

counts <- round(txi$counts)
columnData <- data.frame(condition = sample_table$diabetes_status)
dds <- DESeqDataSetFromMatrix(countData = counts, colData = columnData, design = ~condition) # ~condition is what tells it that I am interested in seeing the relationships based on condition

#Adding the right column names to the dds object
colnames(dds) <- sample_table$sampleName
# colnames(dds)
# dds$condition - These should match with colnames(dds); they do.
```

```{r Sanity check}
# Checking to see that the right columns are associated to the right patient ID

# Get the column names from DESeq object
col_names = colnames(dds)

# Get the patient IDs from the sample table
patient_ids <- sample_table$sampleName

# Compare the column names with the sample name
# If match 
match_result <- match(col_names, patient_ids)

match_result

# Print the results
for (i in seq_along(col_names)) {
  cat("Column:", col_names[i], " - SampleName ID:", patient_ids[match_result[i]], "\n")
}

```

```{r DGE analysis}
#Normalizing the counts

# Estimated size factors are determined by the library size
dds <- estimateSizeFactors(dds) 

# dds$sizeFactor - Pretty cool to see the different relative sizer factors - Need to read about this

# Differential gene expression analysis
dds <- DESeq(dds)

# Getting the results based on the condition of diabetes or not
results <- results(dds, contrast = c("condition", "Yes", "No"))


top_genes <- results[order(results$padj), ]
top_10_genes <- head(top_genes, n=10)

top_10_genes

write.csv(top_10_genes, file = "../Data/to_10_genes.csv")
```
```{r DGE analysis}
#Normalizing the counts

# Estimated size factors are determined by the library size
dds <- estimateSizeFactors(dds) 

# dds$sizeFactor - Pretty cool to see the different relative sizer factors - Need to read about this

# Differential gene expression analysis
dds <- DESeq(dds)

# Getting the results based on the condition of diabetes or not
results <- results(dds, contrast = c("condition", "Yes", "No"))


top_genes <- results[order(results$padj), ]
top_30_genes <- head(top_genes, n=30)

top_30_genes

write.csv(top_30_genes, file = "../Data/top_30_genes_2.csv")
```

```{r}
# Checking to see the what differences is if I sort by smallest p-values and the
# greatest absolute value in log2foldchange

results

# Sort results by absolute log2FC in descending order
sorted_results <- results[order(abs(results$log2FoldChange), decreasing = TRUE), ]

# Print the sorted results
print(sorted_results)

```



```{r}
# Pulling out the results
a = head(results, n = 10)
tail(a, n = 10)
```

```{r}

# This still needs to be properly tested:
# This will automatically pull the ensembl gene description to save

# Load the biomaRt package
library(biomaRt)

# Connect to the Ensembl database
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

# Define the gene identifiers for which you want to retrieve descriptions
gene_ids <- rownames(results)

# Grabbing the first ten for testing
gene_ids = gene_ids[1:10]


### THIS IS WHERE IT DOESNT WORK ###
# Retrieve the gene descriptions using biomaRt
gene_data <- getBM(attributes = c("ensembl_gene_id", "description"), filters = "ensembl_gene_id", values = gene_ids, mart = ensembl)

gene_data
# Save the gene data to a CSV file
write.csv(gene_data, "gene_descriptions.csv", row.names = FALSE)


```


# Checking to see which are different

```{r}
library(dplyr)
dfA = read.csv("../Data/top_30_genes.csv")
dfB = read.csv("../Data/top_30_genes_2.csv")


# Find values in dfA that are not present in dfB
values_only_in_dfA <- anti_join(dfA, dfB, by = "EnsembleID")

# Find values in dfB that are not present in dfA
values_only_in_dfB <- anti_join(dfB, dfA, by = "EnsembleID")

values_only_in_dfA
values_only_in_dfB

```


#Volcano plot
```{r}
library(ggplot2)


# Assuming you have the DESeq results stored in the variable 'results'

# Convert DESeqResults object to data frame
results_df <- as.data.frame(results)

# Calculate the negative log10 of the adjusted p-values
results_df$neg_log10_padj <- -log10(results_df$padj)

# Create a volcano plot
ggplot(results_df, aes(x = log2FoldChange, y = neg_log10_padj)) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results_df, padj < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  labs(x = "log2 Fold Change", y = "-log10(adjusted p-value)",
       title = "Volcano Plot") +
  theme_bw()

```

```{r}
library(ggplot2)

# Assuming you have a DESeqResults object called 'deseq_results'

# Convert DESeqResults to a data frame
results_df <- as.data.frame(results)

results_df$EnsembleID <- rownames(results_df)
# Create the volcano plot
volcano_plot <- ggplot(results_df, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point() +
  xlim(c(-max(abs(results_df$log2FoldChange)), max(abs(results_df$log2FoldChange)))) +
  ylim(c(0, max(-log10(results_df$padj)))) +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  xlab("log2 Fold Change") +
  ylab("-log10(adjusted p-value)") +
  ggtitle("Volcano Plot")

# Add labels to significantly differentially expressed genes
signif_labels <- subset(results_df, padj < 0.05)
volcano_plot_with_labels <- volcano_plot +
  geom_text(data = signif_labels, aes(label = EnsembleID), vjust = 0, hjust = 0)

# Print the volcano plot with labels
print(volcano_plot_with_labels)


```













